socketio/src/client/client.rs

```diff
@@ -322,6 +338,84 @@ mod test {
         Ok(())
     }
 
+    #[test]
+    fn socket_io_reconnect_url_auth_integration() -> Result<()> {
+        fn get_url() -> Url {
+            let timestamp = SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_millis();
+            let mut url = crate::test::socket_io_restart_url_auth_server();
+            url.set_query(Some(&format!("timestamp={timestamp}")));
+            url
+        }
+
+        let connect_num = Arc::new(AtomicUsize::new(0));
+        let close_num = Arc::new(AtomicUsize::new(0));
+        let message_num = Arc::new(AtomicUsize::new(0));
+
+        let connect_num_clone = Arc::clone(&connect_num);
+        let close_num_clone = Arc::clone(&close_num);
+        let message_num_clone = Arc::clone(&message_num);
+
+        let socket = ClientBuilder::new(get_url())
+            .reconnect(true)
+            .max_reconnect_attempts(100)
+            .reconnect_delay(100, 100)
+            .on(Event::Connect, move |_, socket| {
+                connect_num_clone.fetch_add(1, Ordering::SeqCst);
```

Comment:
This can be `Acquire` and `Release´ with the load operation.

Reply:
Does it really matter for the test? The test is based on the existing one which also uses `SeqCst`.

Reply:
I think yes. If you treat your test code badly you will run into trouble sooner or later. Also "someone else made the same mistake" is a bad argument for code IMO.

Reply:
Ok, it can't really be `Release` for the load operation:

```
error: atomic loads cannot have `Release` or `AcqRel` ordering
   --> socketio/src/client/client.rs:480:18
    |
480 |         num.load(Ordering::Release)
    |                  ^^^^^^^^^^^^^^^^^
    |
    = help: consider using ordering modes `Acquire`, `SeqCst` or `Relaxed`
    = note: `#[deny(invalid_atomic_ordering)]` on by default
```

So what should we use?

Reply:
Sorry, load should be `Acquire`, store `Release`.

Reply:
I think it might make the tests simpler to just use a `Mutex<usize>` instead of a `AtomicUsize`.

Reply:
Why? I would always prefer an atomic over a Mutex.

Reply:
Ok, I've changed the orderings.